### PR TITLE
Bug fix for dropped table check constraints when altering a table.

### PR DESCRIPTION
**Pull request by fulghum**
Resolves: https://github.com/dolthub/dolt/issues/2672

Confirmed that alterschema tests now cover all schema properties.

### CODE REVIEW

**Review by timsehn — COMMENTED** [review]
Since we f'ed it up, should we have a bats test as well?

**Comment by zachmu** [comment]
Column name integrity for ALTER should be enforced in the engine, not in dolt. Should be doable.

**Comment by fulghum** [comment]
Added a BATS test; I'll keep digging into how we can enforce integrity for table constraints on schema alter.

File: integration-tests/bats/schema-changes.bats [hunk 1]
```diff
@@ -37,6 +37,16 @@ teardown() {
     [[ ! "$ouput" =~ "Merge failed" ]] || false
 }
 
+@test "schema-changes: dolt schema alter column preserves table checks" {
+    run dolt sql -q "alter table test add constraint test_check CHECK (c2 < 12345);"
+    [ "$status" -eq 0 ]
```
Comment (timsehn) [inline integration-tests/bats/schema-changes.bats:42]: So if all you want to do is check that the command returns 0 just omit `run`. For example 41 and 42 can just be:
`dolt sql -q "alter table test add constraint test_check CHECK (c2 < 12345);"`

`run` populates $status, $output, @lines, etc

**Comment by fulghum** [comment]
Also resolves: https://github.com/dolthub/dolt/issues/2708

**Comment by fulghum** [comment]
Related go-mysq-server PR that adds additional validation to preserve table check constraint integrity when adding or removing columns: https://github.com/dolthub/go-mysql-server/pull/774

**Review by zachmu — APPROVED** [review]
LGTM, but you should also add engine tests in go-mysql-server for the operations where this bug occurred in a follow-up PR.

**Pull request merged**